cpp/src/io/avro/avro.cpp

```diff
@@ -199,9 +199,8 @@ bool container::parse(file_metadata* md, size_t max_num_rows, size_t first_row)
     // Read the next sync markers and ensure they match the first ones we
     // encountered.  If they don't, we have to assume the data is corrupted,
     // and thus, we terminate processing immediately.
-    uint64_t const sync_marker[] = {get_raw<uint64_t>(), get_raw<uint64_t>()};
     bool valid_sync_markers =
-      ((sync_marker[0] == md->sync_marker[0]) && (sync_marker[1] == md->sync_marker[1]));
+      ((get_raw<uint64_t>() == md->sync_marker[0]) && (get_raw<uint64_t>() == md->sync_marker[1]));
```

Comment:
I think we need to keep the variable, because in the new version we might not call the second `get_raw` (and I have to assume there are side effects :))
Could be a `std::array`, if clang-tidy likes that better.

Reply:
Oh that's fun...